### PR TITLE
[Synapse] az synapse spark pool create/update: Add --spark-config-file-path

### PR DESCRIPTION
**Pull request by kevinzz6**
**Description**

Add spark config file path argument in az synapse spark pool create and update cmdlets

- az synapse spark pool create, add argument: --spark-config-file-path

- az synapse spark pool update, add argument: --spark-config-file-path

**Testing Guide**

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Synapse

**Comment by yonzhan** [comment]
Synapse

File: src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py [hunk 1]
```diff
@@ -165,6 +165,9 @@ def load_arguments(self, _):
         # Component Version
         c.argument('spark_version', arg_group='Component Version', help='The supported Spark version is 2.4 now.')
 
+        # Spark config file
+        c.argument('spark_config_file_path', help='Spark pool properties configuration file.')
```
Comment (wonner) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py:169]: >, [](http://example.com/codeflow?start=43&length=1)

Add it under group "Environment Configuration".
`arg_group='Environment Configuration'`

**Review by wonner — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/synapse/manual/_help.py [hunk 2]
```diff
@@ -155,6 +155,10 @@
     text: |-
         az synapse spark pool update --name testpool --workspace-name testsynapseworkspace --resource-group rg \\
         --package-action Add --package package1.jar package2.jar
+  - name: Update the Spark pool's configuration file.
+    text: |-
+        az synapse spark pool update --name testpool --workspace-name testsynapseworkspace --resource-group rg \\
+        --spark-config-file-path 'path/configfile.txt'
```
Comment (zesluo) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/_help.py:161]: @path/configfile? please double check, thanks

Comment (wonner) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/_help.py:161]: Hi @zesluo , I think only JSON file need to add `@` at the front of the path. https://github.com/Azure/azure-cli/blob/dev/doc/quoting-issues-with-powershell.md#best-practice-use-file-input-for-json

File: src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py [hunk 3]
```diff
@@ -1083,12 +1086,14 @@ def test_spark_pool(self):
...
-            self.check('provisioningState', 'Succeeded')
-        ])
+        self.cmd('az synapse spark pool update --ids {pool-id} --tags key1=value1'
+                 ' --spark-config-file-path "{file}"',
+                 checks=[
+                    self.check('tags.key1', 'value1'),
+                    self.check('name', self.kwargs['spark-pool']),
+                    self.check('type', 'Microsoft.Synapse/workspaces/bigDataPools'),
+                    self.check('provisioningState', 'Succeeded')
+                 ])
```
Comment (zesluo) [inline src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py:1097]: no delete spark-config-file-path?

File: src/azure-cli/azure/cli/command_modules/synapse/manual/tests/latest/assets/sparkconfigfile.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+spark.executor.memory 4g
+spark.executor.cores 2
+spark.executor.instances 3
+spark.rpc.message.maxsize 256
+spark.driver.maxResultSize 8g
```
Comment (zesluo) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/tests/latest/assets/sparkconfigfile.txt:5]: we shouldn't have test folder in 'manual' folder?

File: src/azure-cli/azure/cli/command_modules/synapse/manual/operations/sparkpool.py [hunk 5]
```diff
@@ -17,10 +18,9 @@ def get_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_
 def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name,
                       spark_version, node_size, node_count,
                       node_size_family=NodeSizeFamily.memory_optimized.value, enable_auto_scale=None,
-                      min_node_count=None, max_node_count=None,
+                      min_node_count=None, max_node_count=None, spark_config_file_path=None,
                       enable_auto_pause=None, delay=None, spark_events_folder="/events",
                       spark_log_folder="/logs", tags=None, no_wait=False):
-
```
Comment (zesluo) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/operations/sparkpool.py:23]: you can add back the blank line ^_^

**Comment by evelyn-ys** [comment]
Please sync with latest `dev` branch to bypass CI failure

File: src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py [hunk 6]
```diff
@@ -165,6 +165,9 @@ def load_arguments(self, _):
         # Component Version
         c.argument('spark_version', arg_group='Component Version', help='The supported Spark version is 2.4 now.')
 
+        # Spark config file
+        c.argument('spark_config_file_path', arg_group='Environment Configuration', help='Spark pool properties configuration file.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py:169]:
```suggestion
        c.argument('spark_config_file_path', arg_group='Environment Configuration', help='Absolute path of Spark pool properties configuration file.')
```

File: src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py [hunk 7]
```diff
@@ -196,6 +199,9 @@ def load_arguments(self, _):
                    help='Package action must be specified when you add or remove a workspace package from a Apache Spark pool.')
         c.argument('package', arg_group='Custom Libraries', nargs='+', help='List of workspace packages name.')
 
+        # Spark config file
+        c.argument('spark_config_file_path', arg_group='Environment Configuration', help='Spark pool properties configuration file.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py:203]:
```suggestion
        c.argument('spark_config_file_path', arg_group='Environment Configuration', help='Absolute path of Spark pool properties configuration file.')
```

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**